packages/toolpad-components/src/DataGrid.tsx

```diff
@@ -215,7 +231,12 @@ function ImageCell({ field, id, value: src }: GridRenderCellParams<any, any, any
 }
 
 function dateValueGetter({ value }: GridValueGetterParams<any, any>) {
-  return typeof value === 'number' ? new Date(value) : value;
+  try {
+    const date = new Date(value);
+    return date;
+  } catch (err) {
+    return value;
```

Comment:
When does the `Date` constructor throw?

I believe we should implement this as

```tsx
if (value === null || value === undefined || value === '') {
  return value
}
// It's fine if this turns out to be an invalid date, the user wanted a date column, if the data can't parse as a date
// it should just show as such
return new Date(value)
```